aptos-move/aptos-vm/src/move_vm_ext/vm.rs

```diff
@@ -221,11 +225,16 @@ impl MoveVmExt {
             _ => vec![],
         };
 
-        extensions.add(NativeTransactionContext::new(
-            txn_hash.to_vec(),
-            script_hash,
-            self.chain_id,
-        ));
+        if let Some(transaction_context) = transaction_context_opt {
+            extensions.add(NativeTransactionContext::new(
+                txn_hash.to_vec(),
+                script_hash,
```

Comment:
Seems like you can actually store `transaction_context` inside `NativeTransactionContext`? instead of copying each field

Reply:
I refactored it accordingly.